fuzz/zpay32/encode.go

```diff
@@ -0,0 +1,54 @@
+// +build gofuzz
+
+package zpay32fuzz
+
+import (
+	"encoding/hex"
+	"fmt"
+
+	"github.com/btcsuite/btcd/btcec"
+	"github.com/btcsuite/btcd/chaincfg"
+	"github.com/lightningnetwork/lnd/zpay32"
+)
+
+var testMessageSigner zpay32.MessageSigner
+
+func init() {
+	// Initialize the static key we will be using for this fuzz test.
+	testPrivKeyBytes, _ := hex.DecodeString("e126f68f7eafcc8b74f54d269fe206be715000f94dac067d1c04a8ca3b2db734")
```

Comment:
do we really need this in an init method, or could it be moved outside?

Reply:
if we reuse it but since we haven't I'm in favor of moving it back

Reply:
It's just that `init` use is usually discouraged.

Reply:
I changed it, but why?

Reply:
It's usually disallowed by the linter, but we disable that check since it is used for loggers: https://github.com/lightningnetwork/lnd/blob/1558edbc3cee850a6493b8764faa153d7fb09beb/.golangci.yml#L55